mart/attack/perturber.py

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+import torch
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from .projector import Projector
+
+if TYPE_CHECKING:
+    from .initializer import Initializer
+
+__all__ = ["Perturber"]
+
+
+class Perturber(torch.nn.Module):
+    def __init__(
+        self,
+        *,
+        initializer: Initializer,
+        projector: Projector | None = None,
+    ):
+        """_summary_
+
+        Args:
+            initializer (Initializer): To initialize the perturbation.
+            projector (Projector): To project the perturbation into some space.
+        """
+        super().__init__()
+
+        self.initializer_ = initializer
+        self.projector = projector or Projector()
```

Comment:
Is it more consistent with `self.initializer_` if we rename `self.projector` as `self.projector_`?

Reply:
Done in c2f1f77.